File: week2_Arrays/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    int index;
    index = (int)round((double)(0.0588 * letters/words*100 - 0.296 * sentences/words*100 - 15.8));

    if (index<1)
    {
        printf("Before Grade 1\n");
    }
    else if(index>=16)
    {
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", index);
    }
    
}





int count_letters(string text)
{
    int cont=0;
    for(int i=0,length=strlen(text); i<length; i++)
    {
        if((text[i]>='a' && text[i]<='z') || (text[i]>='A' && text[i]<='Z'))
        {
            cont++;
        }
    }
    return cont;
}

int count_words(string text)
{
    int cont=1;
    for(int i=0,length=strlen(text); i<length; i++)
    {
        if(text[i] == ' ')
        {
            cont++;
        }
    }
    return cont;
}

int count_sentences(string text)
{
    int cont=0;
    for(int i=0,length=strlen(text); i<length; i++)
    {
        if(text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            cont++;
        }
    }
    return cont;
}

File: week6_Python/pset6/readability.py
from cs50 import get_int,get_string

def main():
    n = get_string("Text: ")

    l, space, d = 0, 0, 0  # 分别统计字母、空格、句子个数
    for i in n:
        if i.isalpha():
            l = l + 1
        elif i == ' ':
            space = space + 1
        elif i=='.' or i=='!' or i=='?':
            d = d + 1

    result = 0.0588 * l/(space+1)*100 - 0.296 * d/(space+1)*100 - 15.8

    if result<1:
        print("Before Grade 1")
    elif result>=16:
        print("Grade 16+")
    else:
        print("Grade " + str(round(result)))




main()